compatibility_spark-3/src/main/scala/za/co/absa/hyperdrive/compatibility/impl/DeltaCDCToSnapshotWriter.scala

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.hyperdrive.compatibility.impl
+
+import za.co.absa.hyperdrive.compatibility.api.{CompatibleDeltaCDCToSnapshotWriter, DeltaCDCToSnapshotWriterConfiguration}
+import io.delta.tables.{DeltaMergeBuilder, DeltaTable}
+import org.apache.spark.sql.{Column, DataFrame, Row, SaveMode, functions}
+import org.apache.spark.sql.catalyst.expressions.objects.AssertNotNull
+import org.apache.spark.sql.functions.col
+import org.apache.spark.sql.streaming.{OutputMode, StreamingQuery}
+import org.slf4j.LoggerFactory
+
+class DeltaCDCToSnapshotWriter(configuration: DeltaCDCToSnapshotWriterConfiguration) extends CompatibleDeltaCDCToSnapshotWriter {
+  private val logger = LoggerFactory.getLogger(this.getClass)
+
+  override def write(dataFrame: DataFrame): StreamingQuery = {
+    dataFrame.writeStream
+      .trigger(configuration.trigger)
+      .outputMode(OutputMode.Append())
+      .option(configuration.checkpointLocationPropName, configuration.checkpointLocation)
+      .options(configuration.extraConfOptions)
+      .foreachBatch((df: DataFrame, batchId: Long) => {
+        if(!DeltaTable.isDeltaTable(df.sparkSession, configuration.destination)) {
+          df.sparkSession
+            .createDataFrame(df.sparkSession.sparkContext.emptyRDD[Row], df.schema)
+            .write
+            .format("delta")
+            .mode(SaveMode.ErrorIfExists)
```

Comment:
Didn't work for me. It just appended the parquet files to the already existing ones.

Reply:
Yes, I am aware of this behavior. It at least does not remove old files in dir. Other option is to use file system to check if dir is empty, but I wanted to avoid it.